plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/BuildUnsignedGramineEnclave.kt

```diff
@@ -0,0 +1,65 @@
+package com.r3.conclave.plugin.enclave.gradle.gramine
+
+import com.r3.conclave.plugin.enclave.gradle.ConclaveTask
+import com.r3.conclave.plugin.enclave.gradle.div
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.StandardCopyOption
+import javax.inject.Inject
+import kotlin.io.path.absolutePathString
+
+open class BuildUnsignedGramineEnclave @Inject constructor(objects: ObjectFactory) : ConclaveTask() {
+    companion object {
+        const val MANIFEST_TEMPLATE = "bash.manifest.template"
+        const val MANIFEST_DIRECT = "bash.manifest"
+        const val GRAMINE_MANIFEST_EXECUTABLE = "gramine-manifest"
+    }
+
+    @Input
+    val entryPoint: Property<String> = objects.property(String::class.java)
+
+    @Input
+    val archLibDirectory: Property<String> = objects.property(String::class.java)
+
+    @Input
+    val logType: Property<GramineLogType> = objects.property(GramineLogType::class.java)
+
+    @Input
+    val buildDirectory: Property<String> = objects.property(String::class.java)
+
+    @get:OutputFile
+    val outputManifest: RegularFileProperty = objects.fileProperty()
+
+    override fun action() {
+        val templateManifest = copyTemplateManifestToBuildDirectory()
+        generateGramineDirectManifest(templateManifest.absolutePathString())
+    }
+
+    private fun copyTemplateManifestToBuildDirectory(): Path {
+        val outputManifestTemplateFile = Paths.get(buildDirectory.get()) / MANIFEST_TEMPLATE
+
+        BuildUnsignedGramineEnclave::class.java.getResourceAsStream(MANIFEST_TEMPLATE)!!.use {
+            logger.info("Copying $MANIFEST_TEMPLATE to build directory")
+            Files.copy(it, outputManifestTemplateFile, StandardCopyOption.REPLACE_EXISTING)
+        }
+        return outputManifestTemplateFile
+    }
+
+    private fun generateGramineDirectManifest(templateManifest: String) {
+        val command = listOf(
+            GRAMINE_MANIFEST_EXECUTABLE,
+            "-Dlog_level=${logType.get().type}",
```

Comment:
Rather than creating a seperate log level for gramine, instead link it to gradle's logging level. So if we specify info or debug level in gradle then gramine will also do the same. This we can get configure gramine's logging level with no effort, but rather here the logging level has been hardcoded in the plugin class.

Reply:
I can link Gramine log level to Gradle one with a mapping, but my idea is that this needs to be hardcoded in the template manifest and set only to error.
I don't think we expect the manifest generation to fail often, i.e. we should make it so generic that it does never fail. I see the Gramine log something that we do not want to give the user, i.e. I guess we will need to capture and rethrow the exceptions with a nicer message.

Reply:
In that case I don't see any benefit to the Gramine Log Level class, and should be deleted.